src/py/flwr/client/process/clientappio_servicer.py

```diff
@@ -0,0 +1,111 @@
+    PushClientAppOutputsRequest,
+    PushClientAppOutputsResponse,
+)
+from flwr.proto.message_pb2 import Context as ProtoContext
+from flwr.proto.message_pb2 import Message as ProtoMessage
+from flwr.proto.run_pb2 import Run as ProtoRun
+
+
+# pylint: disable=C0103,W0613,W0201
+class ClientAppIoServicer(clientappio_pb2_grpc.ClientAppIoServicer):
+    """ClientAppIo API servicer."""
+
+    def set_object(
+        self,
+        message: Message,
+        context: Context,
+        run: Run,
+        token: int,
+    ) -> None:
+        """Set client app objects."""
+        log(DEBUG, "ClientAppIo.SetObject")
+        # Serialize Message, Context, and Run
+        self.proto_message: ProtoMessage = message_to_proto(message)
+        self.proto_context: ProtoContext = context_to_proto(context)
+        self.proto_run: ProtoRun = run_to_proto(run)
+        self.token: int = token
+
+    def get_object(self) -> Tuple[Message, Context]:
+        """Get client app objects."""
+        log(DEBUG, "ClientAppIo.GetObject")
+        return self.message, self.context
+
+    def _update_object(self) -> None:
+        """Update client app objects."""
+        log(DEBUG, "ClientAppIo.UpdateObject")
+        # Deserialize Message and Context
+        self.message: Message = message_from_proto(self.proto_message)
+        self.context: Context = context_from_proto(self.proto_context)
+
+    def PullClientAppInputs(
+        self, request: PullClientAppInputsRequest, context: grpc.ServicerContext
+    ) -> PullClientAppInputsResponse:
+        """Pull Message, Context, and Run."""
+        log(DEBUG, "ClientAppIo.PullInputs")
+        if request.token != self.token:
+            raise ValueError("Mismatch between ClientApp and SuperNode token")
+        return PullClientAppInputsResponse(
+            message=self.proto_message,
+            context=self.proto_context,
+            run=self.proto_run,
+        )
+
+    def PushClientAppOutputs(
+        self, request: PushClientAppOutputsRequest, context: grpc.ServicerContext
+    ) -> PushClientAppOutputsResponse:
+        """Push Message and Context."""
+        log(DEBUG, "ClientAppIo.PushOutputs")
+        if request.token != self.token:
+            raise ValueError("Mismatch between ClientApp and SuperNode token")
+        self.proto_message = request.message
+        self.proto_context = request.context
+        # Update Message and Context
+        try:
+            self._update_object()
+            # Set status
+            code = typing.ClientAppOutputCode.SUCCESS
+            status = typing.ClientAppOutputStatus(code=code, message="Success")
+            proto_status = clientappstatus_to_proto(status=status)
+            return PushClientAppOutputsResponse(status=proto_status)
+        except Exception as e:  # pylint: disable=broad-exception-caught
+            log(ERROR, "ClientApp failed to push message to SuperNode, %s", e)
+            return PushClientAppOutputsResponse()
```

Comment:
Should this `PushClientAppOutputsResponse()` return some `status` that we predefine here in the body of the `except`?

Reply:
Good point, added.